packages/ast-spec/src/unions/Node.ts

```diff
@@ -0,0 +1,325 @@
+import type { TSProtectedKeyword } from '../token/TSProtectedKeyword/spec';
+import type { TSPublicKeyword } from '../token/TSPublicKeyword/spec';
+import type { TSReadonlyKeyword } from '../token/TSReadonlyKeyword/spec';
+import type { TSStaticKeyword } from '../token/TSStaticKeyword/spec';
+import type { TSAnyKeyword } from '../type/TSAnyKeyword/spec';
+import type { TSArrayType } from '../type/TSArrayType/spec';
+import type { TSBigIntKeyword } from '../type/TSBigIntKeyword/spec';
+import type { TSBooleanKeyword } from '../type/TSBooleanKeyword/spec';
+import type { TSConditionalType } from '../type/TSConditionalType/spec';
+import type { TSConstructorType } from '../type/TSConstructorType/spec';
+import type { TSFunctionType } from '../type/TSFunctionType/spec';
+import type { TSImportType } from '../type/TSImportType/spec';
+import type { TSIndexedAccessType } from '../type/TSIndexedAccessType/spec';
+import type { TSInferType } from '../type/TSInferType/spec';
+import type { TSIntersectionType } from '../type/TSIntersectionType/spec';
+import type { TSLiteralType } from '../type/TSLiteralType/spec';
+import type { TSMappedType } from '../type/TSMappedType/spec';
+import type { TSNamedTupleMember } from '../type/TSNamedTupleMember/spec';
+import type { TSNeverKeyword } from '../type/TSNeverKeyword/spec';
+import type { TSNullKeyword } from '../type/TSNullKeyword/spec';
+import type { TSNumberKeyword } from '../type/TSNumberKeyword/spec';
+import type { TSObjectKeyword } from '../type/TSObjectKeyword/spec';
+import type { TSOptionalType } from '../type/TSOptionalType/spec';
+import type { TSParenthesizedType } from '../type/TSParenthesizedType/spec';
+import type { TSQualifiedName } from '../type/TSQualifiedName/spec';
+import type { TSRestType } from '../type/TSRestType/spec';
+import type { TSStringKeyword } from '../type/TSStringKeyword/spec';
+import type { TSSymbolKeyword } from '../type/TSSymbolKeyword/spec';
+import type { TSTemplateLiteralType } from '../type/TSTemplateLiteralType/spec';
+import type { TSThisType } from '../type/TSThisType/spec';
+import type { TSTupleType } from '../type/TSTupleType/spec';
+import type { TSTypeLiteral } from '../type/TSTypeLiteral/spec';
+import type { TSTypeOperator } from '../type/TSTypeOperator/spec';
+import type { TSTypePredicate } from '../type/TSTypePredicate/spec';
+import type { TSTypeQuery } from '../type/TSTypeQuery/spec';
+import type { TSTypeReference } from '../type/TSTypeReference/spec';
+import type { TSUndefinedKeyword } from '../type/TSUndefinedKeyword/spec';
+import type { TSUnionType } from '../type/TSUnionType/spec';
+import type { TSUnknownKeyword } from '../type/TSUnknownKeyword/spec';
+import type { TSVoidKeyword } from '../type/TSVoidKeyword/spec';
+import type { Literal } from './Literal';
+
+/*
+ * NOTE:
+ * Tokens are not included in the `Node` union below on purpose because they are not ever included as part of the standard AST tree.
+ */
+
+export type Node =
```

Comment:
q: can't we just import nodes from unions instead of importing them all again?

Reply:
that makes the assumption that every node is part of a union.

Also it's impossible to tell if a type like this contains every node
```ts
type Node = Statement | Expression | ...
```

OTOH it's trivial to tell if this union type contains every node - just ctrl+f